File: stone_lib/src/main/java/com/github/klee0kai/stone/closed/provide/ProvideConsumer.java
package com.github.klee0kai.stone.closed.provide;

import com.github.klee0kai.stone.closed.types.StListUtils;

import java.util.Collection;
import java.util.LinkedList;

public class ProvideConsumer<T> {

    private LinkedList<T> list = new LinkedList<>();

    public void addAll(Collection<T> collection) {
        if (collection != null) {
            list.addAll(StListUtils.filter(collection, (i, it) -> it != null));
        }
    }

    public void add(T element) {
        if (element != null) {
            list.add(element);
        }
    }

    public LinkedList<T> getList() {
        return list;
    }

    public T getFirst() {
        return list.isEmpty() ? null : list.getFirst();
    }


}

File: stone_multiplatform/src/commonMain/kotlin/com/github/klee0kai/stone/__hidden__/provide/ProvideConsumer.kt
package com.github.klee0kai.stone.__hidden__.provide

class ProvideConsumer<T> {
    private val _list = mutableListOf<T>()

    val list get() = _list.toList()

    val first: T? get() = _list.firstOrNull()

    fun addAll(
        collection: Collection<T?>
    ) {
        _list.addAll(collection.filterNotNull())
    }

    fun add(element: T?) {
        if (element != null) {
            _list.add(element)
        }
    }

}
